com.unity.netcode.adapter.utp/Runtime/NetworkMetricsPipelineStage.cs

```diff
@@ -0,0 +1,69 @@
+#if MULTIPLAYER_TOOLS
```

Comment:
Isn't this redundant with the check below?

Reply:
It is yes. But it's also clear that you need TOOLS PRE 4 or higher. Eventually we will remove that check probably and I wouldn't want us to forget to put the TOOLS check. But if you rather have just the TOOLS_PRE_4, I can do that

Reply:
I don't have any strong opinion on the matter, honestly. If the pre.4 check is meant to go away at some point, then I agree leaving the more generic tools check is safer.